src/index.js

```diff
@@ -53,45 +54,56 @@ function createBabelPluginFactory(customCallback = returnObject) {
 			externalHelpersWhitelist,
 			include,
 			runtimeHelpers,
+			transformGenerated,
 			...babelOptions
 		} = unpackOptions(pluginOptions);
 
+		if (transformGenerated) {
+			return {
+				name: 'babel-output',
+
+				renderStart() {
+					if (pluginOptions.include || pluginOptions.exclude || pluginOptions.extensions) {
+						warnOnce(
```

Comment:
Would prefer moving this warning out of this render hook, this can be validated just after entering transformGenerated branch of code

Reply:
The problem is that I do not have access to `this.warn` at that point, so all I could do is log to the console or throw a real error, which did not seem ideal to me. `renderStart` will be the first hook that runs and would be the canonical place to throw warnings due to configuration issues, at least for output plugins. For input plugins, it would be `buildStart`.

Reply:
Ok, makes sense 👍